corehq/apps/reports/standard/monitoring.py

```diff
@@ -886,40 +887,43 @@ def _get_rows(self):
                 mobile_user_and_group_slugs,
             )
 
-            placeholders = []
-            params = []
             user_map = {user.user_id: user
                         for user in users_data.combined_users if user.user_id}
+            user_ids = [user.user_id for user in users_data.combined_users if user.user_id]
+
+            xmlnss = []
+            app_ids = []
             form_map = {}
+
             for form in self.all_relevant_forms.values():
-                placeholders.append('(%s,%s)')
-                params.extend([form['app_id'], form['xmlns']])
-                form_map[form['xmlns']] = form
-
-            where = '(app_id, xmlns) in (%s)' % (','.join(placeholders))
-            results = FormData.objects \
-                .filter(received_on__range=(self.datespan.startdate_utc, self.datespan.enddate_utc)) \
-                .filter(user_id__in=user_map.keys()) \
-                .values('instance_id', 'user_id', 'time_end', 'received_on', 'xmlns')\
-                .extra(
-                    where=[where], params=params
-                )
-            if results.count() > 5000:
```

Comment:
it automatically does when we raise the `TooMuchDataError` which gets done here: https://github.com/dimagi/commcare-hq/pull/10368/files#diff-30c4e1ef0d3ceb995aab05246948b579R1029